### ISSUE TITLE
InstanceOf matcher does not accept null values -> mockedObject.setValue(null) is not possible

### ISSUE THREAD

**Issue by Martin-Wegner** [mockito/mockito#726]
I don't know if this is Mockito or GwtMockito related (https://github.com/google/gwtmock
https://github.com/google/gwtmockito/issues/62). I mocked a text field and used an answer object to connect the `getValue()` and `setValue()` methods. Both methods work with a string. This worked fine with Mockito 1.10.19 but since Mockito 2.1.0 the following case is broken: `textField.setValue(null);`

I debugged the code and saw that the answer object is never been called. The reason is, that the `InstanceOf.matches()` method does not accept a null value:
```
public boolean matches(Object actual) {
    return (actual != null) &&
            (Primitives.isAssignableFromWrapper(actual.getClass(), clazz)
                    || clazz.isAssignableFrom(actual.getClass()));
}
```
`textField.setValue(null);` -> `actual` is null and therefore the answer object is not called.

In other words: It is not possible to set a null string.

My answer implementation:
```
public class ValueAnswer implements Answer<Object> {

  private Object value;

  @Override
  public Object answer(final InvocationOnMock invocation) throws Throwable {

    if ((invocation != null) && (invocation.getArguments() != null) && (invocation.getArguments().length == 1)) {
      value = invocation.getArguments()[0];
    }

    return value;
  }

}
````

**Comment by ChristianSchwarz** [comment]
@Martin-Wegner can you provide your test? 

Note: Better use[ AdditionalAnswers.returnsFirstArg()](http://site.mockito.org/mockito/docs/current/org/mockito/AdditionalAnswers.html#returnsFirstArg())

**Comment by Martin-Wegner** [comment]
@ChristianSchwarz thanks!

The new Mockito 2 API says:

> Since Mockito 2.1.0, only allow non-null String. As this is a nullable reference, the suggested API to match null wrapper would be isNull(). We felt this change would make tests harness much safer that it was with Mockito 1.x.

So I changed `ArgumentMatchers.anyString()` to `ArgumentMatchers.any()`.

**Comment by bric3** [comment]
@Martin-Wegner Note that `any()` matches anything, `String` or not, that may be ok if the signature constrain the argument type well enough (which seems to be the case). In other case and if needed type aware matchers have to be used.

**Comment by Martin-Wegner** [comment]
@bric3 thanks for your reply. Yes I think in my case it is not a problem. Is there an `or` operator for matchers?

**Comment by bric3** [comment]
@Martin-Wegner Yes there is in `AdditionalMatchers`
